modules/rosetta/src/main/scala/org/tessellation/rosetta/http/routes/ConstructionRoutes.scala

```diff
@@ -57,6 +57,25 @@ final class ConstructionRoutes[F[_]: Async](
             .asRosettaResponse
         }
         .handleUnknownError
+
+    case req @ POST -> Root / "combine" =>
+      req
+        .decodeRosettaWithNetworkValidation[ConstructionCombine.Request](appEnvironment, _.networkIdentifier) { combineReq =>
+          val response = NonEmptyList
+            .fromList(combineReq.signatures)
+            .flatMap { signatures =>
+              signatures match {
+                case _ :: Nil =>
+                  constructionService
+                    .combineTransaction(combineReq.unsignedTransaction, signatures)
+                    .some
+                case _ => none
+              }
+            }
+            .map(_.bimap(_.toRosettaError, ConstructionCombine.Response(_)).asRosettaResponse.handleUnknownError)
+
+          response.get
```

Comment:
if you want to use `get`, you should use `getOrElse` providing the value for empty (none) case.
anyway, you can type `combineReq.signatures` as NonEmptyList, so then you can just pass it to the construction service and map its output, without additional logic here in the routes

Reply:
I reworked the logic to not need this.